modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/ShelfRearrangeActionsPage.kt

```diff
@@ -0,0 +1,95 @@
+package au.com.shiftyjelly.pocketcasts.player.view.shelf
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.lifecycle.asFlow
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import au.com.shiftyjelly.pocketcasts.compose.bars.ThemedTopAppBar
+import au.com.shiftyjelly.pocketcasts.compose.theme
+import au.com.shiftyjelly.pocketcasts.player.viewmodel.PlayerViewModel
+import au.com.shiftyjelly.pocketcasts.player.viewmodel.ShelfViewModel
+import au.com.shiftyjelly.pocketcasts.preferences.model.ShelfItem
+import au.com.shiftyjelly.pocketcasts.ui.helper.ColorUtils
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import au.com.shiftyjelly.pocketcasts.ui.theme.ThemeColor
+import kotlinx.coroutines.flow.distinctUntilChangedBy
+import kotlinx.coroutines.flow.map
+import android.graphics.Color as AndroidColor
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+@Composable
+fun ShelfRearrangeActionsPage(
+    theme: Theme,
+    shelfViewModel: ShelfViewModel,
+    playerViewModel: PlayerViewModel,
+    onBackPressed: () -> Unit,
+) {
+    val iconColorInt = ThemeColor.playerContrast01(theme.activeTheme)
+    val backgroundColorInt = theme.playerBackground2Color(playerViewModel.podcast)
+    val toolbarColorInt = theme.playerBackgroundColor(playerViewModel.podcast)
+    val selectedColorInt = theme.playerHighlight7Color(playerViewModel.podcast)
+    val selectedBackgroundInt = ColorUtils.calculateCombinedColor(backgroundColorInt, selectedColorInt)
+
+    val shelfItems by playerViewModel.shelfLive.asFlow()
+        .collectAsStateWithLifecycle(emptyList<ShelfItem>())
+
+    val episode by playerViewModel.playingEpisodeLive.asFlow()
+        .map { (episode, _) -> episode }
+        .distinctUntilChangedBy { it.uuid }
+        .collectAsStateWithLifecycle(null)
+
+    shelfViewModel.setData(shelfItems, episode)
```

Comment:
This should be wrapped in a `LaunchedEffect`.

Reply:
Right 🤦‍♀️ 

[fcc3ae4](https://github.com/Automattic/pocket-casts-android/pull/3215/commits/fcc3ae4c8bc863912b721d3f8255fb259fbc2750)